include/mfmg/dealii/dealii_mesh_evaluator.hpp

```diff
@@ -58,9 +60,9 @@ class DealIIMeshEvaluator : public MeshEvaluator
     ASSERT_THROW_NOT_IMPLEMENTED();
   }
 
-  dealii::DoFHandler<dim> &get_dof_handler();
+  dealii::DoFHandler<dim> &get_dof_handler() const;
```

Comment:
Isn't that kind of strange to have a `const` function that return a `non-const` reference to protected data ?

Reply:
Good point.  Let me revisit that one.

Reply:
Considering that `DealIIMeshEvaluator::evaluate_{agglomerate, global}()` require non-`const` reference to the dof handler and the constraints, I think the right thing to do is revert my marking `get_dof_handler()` and `get_constraints()` as `const` member function and have `get_diagonal()` just not be `const`-qualified.

Reply:
That's fine.